### PR TITLE
Add Riot.js

### PR DESCRIPTION
**Pull request by GianlucaGuarini**
## Description
Add Riot.js using the `.riot` format as recommended by @Alhadis [here](https://github.com/riot/riot/issues/2529#issuecomment-379290366)
This PR fixes the issues identified in the previous one https://github.com/github/linguist/pull/3720

## Checklist:
- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?q=extension%3Ariot+NOT+nothack&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/riot/examples/
    - Sample license(s): https://github.com/riot/examples/blob/gh-pages/LICENSE
  
  - [x] I have included a syntax highlighting grammar: https://github.com/riot/syntax-highlight
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: samples/Riot/sample.riot [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      }
+    }
+  </script>
+
+  <style>
+    :host { font-size: 2rem }
+    h3 { color: #444 }
+    ul { color: #999 }
+  </style>
+</sample>
```
Comment (lildude) [inline samples/Riot/sample.riot]: This looks like a standard "Hello world" sample. Can you please replace it with a more realistic real-world sample.

Comment (GianlucaGuarini) [reply samples/Riot/sample.riot]: Sure no problem I will do it asap. Thank you for your feedback

Comment (GianlucaGuarini) [reply samples/Riot/sample.riot]: I have updated the template of the PR and added new "real world" examples from the official examples

**Comment by GianlucaGuarini** [comment]
@lildude Please let me know if I may I help you here on anything. I think it should be everything fine now

**Pull request merged**